garage/tf/policies/gaussian_lstm_policy_with_model.py

```diff
@@ -0,0 +1,295 @@
+            forget_bias=forget_bias,
+            layer_normalization=layer_normalization,
+            learn_std=learn_std,
+            std_share_network=std_share_network,
+            init_std=init_std)
+
+        self._initialize()
+
+    def _initialize(self):
+        obs_ph = tf.placeholder(
+            tf.float32, shape=(None, None, self._input_dim))
+        step_input_var = tf.placeholder(
+            shape=(None, self._input_dim), name='step_input', dtype=tf.float32)
+        step_hidden_var = tf.placeholder(
+            shape=(None, self._hidden_dim),
+            name='step_hidden_input',
+            dtype=tf.float32)
+        step_cell_var = tf.placeholder(
+            shape=(None, self._hidden_dim),
+            name='step_cell_input',
+            dtype=tf.float32)
+
+        with tf.variable_scope(self.name) as vs:
+            self._variable_scope = vs
+            self.model.build(obs_ph, step_input_var, step_hidden_var,
+                             step_cell_var)
+
+        self._f_step_mean_std = tf.get_default_session().make_callable(
+            [
+                self.model.networks['default'].sample,
+                self.model.networks['default'].step_mean,
+                self.model.networks['default'].step_log_std,
+                self.model.networks['default'].step_hidden,
+                self.model.networks['default'].step_cell
+            ],
+            feed_list=[step_input_var, step_hidden_var, step_cell_var])
+
+        self.prev_actions = None
+        self.prev_hiddens = None
+        self.prev_cells = None
+
+    @property
+    def vectorized(self):
+        """Vectorized or not."""
+        return True
+
+    def dist_info_sym(self, obs_var, state_info_vars, name=None):
+        """
+        Symbolic graph of the distribution.
+
+        Args:
+            obs_var (tf.Tensor): Tensor input for symbolic graph.
+            state_info_vars (dict): Extra state information, e.g.
+                previous action.
+            name (str): Name for symbolic graph.
+
+        """
+        if self._state_include_action:
+            prev_action_var = state_info_vars['prev_action']
+            prev_action_var = tf.cast(prev_action_var, tf.float32)
+            all_input_var = tf.concat(
+                axis=2, values=[obs_var, prev_action_var])
+        else:
+            all_input_var = obs_var
+
+        with tf.variable_scope(self._variable_scope):
+            _, mean_var, _, log_std_var, _, _, _, _, _, _ = self.model.build(
+                all_input_var,
+                self.model.networks['default'].step_input,
+                self.model.networks['default'].step_hidden_input,
+                self.model.networks['default'].step_cell_input,
+                name=name)
+
+        return dict(mean=mean_var, log_std=log_std_var)
+
+    def reset(self, dones=None):
+        """
+        Reset the policy.
+
+        Args:
+            dones (numpy.ndarray): Bool that indicates terminal state(s).
```

Comment:
it seems this need to be a numpy.ndarray of bools, right?

Reply:
Yes, do you mean we should state `dones (numpy.ndarray[Bool])`?

Reply:
meh, on second thought let's not go this deep with the typing yet.